tests/sentry/web/frontend/test_auth_organization_login.py

```diff
@@ -997,3 +936,111 @@ def test_correct_redirect_as_2fa_user_invited_member(self):
         )
 
         assert resp.redirect_chain == [("/auth/2fa/", 302)]
+
+
+class OrganizationAuthLoginNoPasswordTest(AuthProviderTestCase):
+    def setUp(self):
+        self.owner = self.create_user()
+        self.organization = self.create_organization(name="foo", owner=self.owner)
+        self.user = self.create_user("bar@example.com", is_managed=False, password="")
+        self.auth_provider = AuthProvider.objects.create(
+            organization=self.organization, provider="dummy"
+        )
+        self.path = reverse("sentry-auth-organization", args=[self.organization.slug])
+        self.auth_sso_path = reverse("sentry-auth-sso")
+        UserEmail.objects.filter(user=self.user, email="bar@example.com").update(is_verified=False)
+
+    def test_flow_authenticated_without_verified_without_password(self):
+        """
+        Given an existing authenticated user, and an updated identity (e.g.
+        the ident changed from the SSO provider), we should be re-linking
+        the identity automatically as they don't have a password.
+
+        This is specifically testing an unauthenticated flow.
+        """
+
+        # setup a 'previous' identity, such as when we migrated Google from
+        # the old idents to the new
+        assert not self.user.has_usable_password()
+        self.create_member(organization=self.organization, user=self.user)
+
+        resp = self.client.post(self.path, {"init": True})
+
+        assert resp.status_code == 200
+        assert self.provider.TEMPLATE in resp.content.decode("utf-8")
+
+        resp = self.client.post(self.auth_sso_path, {"email": "bar@example.com"})
+        self.assertTemplateUsed(resp, "sentry/auth-confirm-identity.html")
+        assert resp.status_code == 200
+        assert resp.context["existing_user"] is None
+
+    @with_feature("organizations:idp-automatic-migration")
+    @mock.patch("sentry.auth.helper.send_one_time_account_confirm_link")
+    def test_flow_automatically_migrated_without_verified_without_password(
+        self, mock_send_one_time_account_confirm_link
+    ):
+        # setup a 'previous' identity, such as when we migrated Google from
```

Comment:
again, not sure what this comment means. we aren't setting up a previous identity here?

Reply:
I think the word "identity" is a little overloaded. We are not setting up an identity as such, but we could clarify the comment to say that we are setting up the state that would have been left behind by auto-provisioning from a now-deleted `AuthIdentity`.

Reply:
i think the comment is just errant and copy pasted. in the other tests we do set up an AuthIdentity right after that line.